microservices/optimizer/app.py

```diff
@@ -38,8 +46,38 @@ def update_times() -> tuple[Response, int]:
     associated with the repository along with the execution times of each test
     classes of a recent deployment. The optimizer service should add this
     execution time into the database for the given test classes."""
-    app.logger.info("update endpoint invoked")
-    return jsonify({"error": "unimplemented"}), 501
+    data = request.json
+    if data is None:
+        return jsonify({"error": "missing request body"}), 400
+
+    url = data.get("url")
+    branch = data.get("branch")
+    times = data.get("times")
+
+    if not url or not branch or times is None:
+        return jsonify({"error": "missing or invalid parameters"}), 400
+
+    try:
+        for test_class in times:
+            name = test_class.get("name")
+            time = test_class.get("time")
+            if name is None or time is None:
+                return (
+                    jsonify({"error": "missing test class name or execution time"}),
+                    400,
+                )
+
+            db_manager: DBManager | None = getattr(g, "db_manager", None)
+            if db_manager is not None:
```

Comment:
If `db_manager` is `None`, we should probably return an error code since we can't get a database connection (should be a 500 internal error)